packages/frontend/src/components/ElectionForm/Races/RaceForm.tsx

```diff
@@ -11,64 +11,98 @@ import { Box, FormHelperText, Radio, RadioGroup, Stack } from "@mui/material"
 import IconButton from '@mui/material/IconButton'
 import ExpandLess from '@mui/icons-material/ExpandLess'
 import ExpandMore from '@mui/icons-material/ExpandMore'
-// import { scrollToElement } from '../../util';
+import { scrollToElement } from '../../util';
 import useElection from '../../ElectionContextProvider';
 import { v4 as uuidv4 } from 'uuid';
 import useConfirm from '../../ConfirmationDialogProvider';
 import useFeatureFlags from '../../FeatureFlagContextProvider';
+import { use } from 'i18next';
+import { on } from 'events';
 
 export default function RaceForm({ race_index, editedRace, errors, setErrors, applyRaceUpdate }) {
     const flags = useFeatureFlags();
     const [showsAllMethods, setShowsAllMethods] = useState(false)
     const { election } = useElection()
-
     const confirm = useConfirm();
+    const inputRefs = useRef([]);
+    const ephemeralCandidates = useMemo(() => 
+        [...editedRace.candidates, { candidate_id: uuidv4(), candidate_name: '' }], 
+        [editedRace.candidates]
+    );    // BottomRef creates a reference to the bottom of the window to scroll to when a new candidate is added
+    const bottomRef = useRef(null);
+    useEffect(() => {
+            bottomRef.current.scrollIntoView(false)
+    }, [ephemeralCandidates])
 
-    const onEditCandidate = (candidate: Candidate, index) => {
-        setErrors({ ...errors, candidates: '', raceNumWinners: '' })
+    const onEditCandidate = useCallback((candidate, index) => {
```

Comment:
Neat, I haven't used useCallback much before, this looks like a good use case

Reply:
Yeah it was getting bogged down from having to be recreated on every key stroke, and it's not too much effort to have it useCallback.